File: packages/app/src/services/software/definitions/python.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'python';
	public name = 'Python';
	public category = [SoftwareCategory.Development];
	public downloadName = '';
	public icon = 'python.png';
	public homepage = 'https://python.org';

	public async resolveDownloadUrl() {
		return ok('');
	}
}

File: src/lib/software/python.py
from src.lib.software.variants.python_313 import Python313
from src.lib.software.variants.python_312 import Python312
from src.lib.software import BaseSoftware, SoftwareCategory

class Python(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.key = 'python'
        self.name = 'Python'
        self.category = [SoftwareCategory.Development]
        self.variants = [Python313(), Python312()]
        self.icon = 'python.png'
        self.homepage = 'https://python.org'

    def resolve_download_url(self):
        pass
